test/AxelarGateway.js

```diff
@@ -671,276 +828,409 @@ describe('AxelarGateway', () => {
+                    threshold,
+                    operators.slice(0, threshold),
+                );
 
-    describe('command transferOperatorship', () => {
-        beforeEach(async () => {
-            await deployGateway();
+                const tx = await gateway.execute(input, getGasOptions());
+
+                await expect(tx)
+                    .to.emit(gateway, 'Executed')
+                    .withArgs(commandIDs[0])
+                    .and.to.emit(gateway, 'Executed')
+                    .withArgs(commandIDs[1])
+                    .and.to.emit(externalToken, 'Transfer')
+                    .withArgs(depositHandlerAddress, gateway.address, burnAmount);
+
+                return externalToken
+                    .balanceOf(depositHandlerAddress)
+                    .then(bigNumberToNumber)
+                    .then((balance) => {
+                        expect(balance).to.eq(0);
+                    })
+                    .then(async () => await externalToken.transfer(depositHandlerAddress, burnAmount).then((tx) => tx.wait()))
+                    .then(async () => {
+                        const commandID = getRandomID();
+                        const input = await getSignedWeightedExecuteInput(
+                            buildCommandBatch(await getChainId(), [commandID], ['burnToken'], [getBurnCommand(externalSymbol, salt)]),
+                            operators,
+                            getWeights(operators),
+                            threshold,
+                            operators.slice(0, threshold),
+                        );
+
+                        const tx = await gateway.execute(input, getGasOptions());
+                        await expect(tx)
+                            .to.emit(gateway, 'Executed')
+                            .withArgs(commandID)
+                            .and.to.emit(externalToken, 'Transfer')
+                            .withArgs(depositHandlerAddress, gateway.address, burnAmount);
+
+                        return await externalToken.balanceOf(depositHandlerAddress);
+                    })
+                    .then(bigNumberToNumber)
+                    .then((balance) => {
+                        expect(balance).to.eq(0);
+                    });
+            });
         });
 
-        it('should allow operators to transfer operatorship', async () => {
-            const newOperators = ['0xb7900E8Ec64A1D1315B6D4017d4b1dcd36E6Ea88', '0x6D4017D4b1DCd36e6EA88b7900e8eC64A1D1315b'];
+        describe('negative test cases', () => {
+            before(async () => {
+                await deployGateway();
+            });
 
-            const data = buildCommandBatch(
-                await getChainId(),
-                [getRandomID()],
-                ['transferOperatorship'],
-                [getTransferWeightedOperatorshipCommand(newOperators, getWeights(newOperators), newOperators.length)],
-            );
+            before(async () => {
```

Comment:
maybe create a helper under this test? only difference with positive test is the different factory

Reply:
any reason to not refactor this into a helper?

Reply:
I can add a quick fix to refactor the common code into a helper function